src/mongocxx/events/command_failed_event.cpp

```diff
@@ -52,6 +56,15 @@ std::int64_t command_failed_event::operation_id() const {
         static_cast<const mongoc_apm_command_failed_t*>(_failed_event));
 }
 
+bsoncxx::stdx::optional<bson_oid_t> command_failed_event::service_id() const {
+    if(const bson_oid_t *result = libmongoc::apm_command_failed_get_service_id(
```

Comment:
It is indeed. Since C++17, if-statements (and most other conditionals) can have an init-specifier, similar to a for-statement:
if(init; cond) {} else { /* init is still in scope */ }